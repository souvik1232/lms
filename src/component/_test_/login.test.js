import React from 'react';
import {shallow} from 'enzyme';
import Login from "../login/login" 

describe('<Login /> with no props', () => {
  const container = shallow(<Login />);
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });
});

describe('Login Component', () => {
  //testing rendering of login component
    it('should render without throwing an error', () => {
        expect(shallow(<Login/>).exists()).toBe(true)
    })
   // testing the email and password input existence
    it('renders a email input', () => {
        expect(shallow( <Login/> ).find('#outlined-basic').length).toEqual(1)
      })
      it('renders a password input', () => {
        expect(shallow( <Login/> ).find('#outlined-basic1').length).toEqual(1)
      })
      test("it renders placeholder text same as passed from props",()=>{
        const placeholderProps="test"
       const wrapper=shallowExample({placeholder:placeholderProps})
    //    console.log("wrapper details",wrapper.debug());
       expect(wrapper.find('input').at(0).props().placeholder).toEqual(placeholderProps)
      })
})

describe("Button Component Tests", () => {
  it("Renders correctly in DOM", () => {
      shallow(
          <button text="Test" />
      );
  });
  it("Expects to find button HTML element in the DOM", () => {
      const wrapper = shallow(<button text="test"/>)
      expect(wrapper.find('button')).toHaveLength(1);
  });

  it("Expects to find button HTML element with className pgn in the DOM", () => {
      const wrapper = shallow(<button className="pgn" text="test"/>)
      expect(wrapper.find('button.pgn')).toHaveLength(1);
  });

  it("Expects to run onClick function when button is pressed in the DOM", () => {
      const mockCallBackClick = jest.fn();
      const wrapper = shallow(<button onClick={mockCallBackClick} className="btn" text="test"/>);
      wrapper.find('button').simulate('click');
      expect(mockCallBackClick.mock.calls.length).toEqual(1);
  });
})

describe('counter testing ',()=>{
  test('render the title ', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.find('.hdr1').text()).toContain("Welcome back!");
  })
})
describe('counter testing ',()=>{
  test('render the title ', () => {
    const wrapper = shallow(<Login/>);
    expect(wrapper.find('.hdr2').text()).toContain("Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumyundefined");
  })
})
