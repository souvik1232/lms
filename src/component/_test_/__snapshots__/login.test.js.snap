// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> with no props should match the snapshot 1`] = `"<div class=\\"login\\"><div class=\\"img-container\\"><span class=\\" lazy-load-image-background blur\\" style=\\"background-image:;background-size:;display:inline-block\\"><span class=\\"img1\\" style=\\"display:inline-block\\"></span></span></div><div class=\\"inputs\\"><img class=\\"l1\\" alt=\\"\\"/><br/><span class=\\"hdr1\\">Welcome back!</span><br/><span class=\\"hdr2\\">Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed <br/>diam nonumyundefined</span><br/><br/><br/><div class=\\"inpt1\\"><div class=\\"MuiFormControl-root MuiTextField-root\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-basic\\" id=\\"outlined-basic-label\\">Email</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" id=\\"outlined-basic\\" class=\\"MuiInputBase-input MuiOutlinedInput-input\\"/><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Email</span></legend></fieldset></div></div></div><br/><br/><br/> <br/><div class=\\"inpt1\\"><div class=\\"MuiFormControl-root MuiTextField-root passfield\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined\\" data-shrink=\\"false\\" for=\\"outlined-basic1\\" id=\\"outlined-basic1-label\\">Password</label><div class=\\"MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd\\"><input type=\\"password\\" aria-invalid=\\"false\\" id=\\"outlined-basic1\\" class=\\"MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd\\"/><div class=\\"MuiInputAdornment-root MuiInputAdornment-positionEnd\\"><button class=\\"MuiButtonBase-root MuiIconButton-root Visibility MuiIconButton-edgeEnd\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"toggle password visibility\\"><span class=\\"MuiIconButton-label\\"><svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z\\"></path></svg></span></button></div><fieldset aria-hidden=\\"true\\" class=\\"PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline\\"><legend class=\\"PrivateNotchedOutline-legendLabelled-3\\"><span>Password</span></legend></fieldset></div></div></div><br/><br/><br/><span class=\\"frgt\\">Forgot Password?</span><br/><br/><button class=\\"loginbut\\">Login</button></div></div>"`;
